src/cascadia/TerminalApp/AboutDialog.cpp

```diff
@@ -0,0 +1,125 @@
+    }
+
+    void AboutDialog::_SendFeedbackOnClick(const IInspectable& /*sender*/, const Windows::UI::Xaml::Controls::ContentDialogButtonClickEventArgs& /*eventArgs*/)
+    {
+#if defined(WT_BRANDING_RELEASE)
+        ShellExecute(nullptr, nullptr, L"https://go.microsoft.com/fwlink/?linkid=2125419", nullptr, nullptr, SW_SHOW);
+#else
+        ShellExecute(nullptr, nullptr, L"https://go.microsoft.com/fwlink/?linkid=2204904", nullptr, nullptr, SW_SHOW);
+#endif
+    }
+
+    void AboutDialog::_ThirdPartyNoticesOnClick(const IInspectable& /*sender*/, const Windows::UI::Xaml::RoutedEventArgs& /*eventArgs*/)
+    {
+        std::filesystem::path currentPath{ wil::GetModuleFileNameW<std::wstring>(nullptr) };
+        currentPath.replace_filename(L"NOTICE.html");
+        ShellExecute(nullptr, nullptr, currentPath.c_str(), nullptr, nullptr, SW_SHOW);
+    }
+
+    bool AboutDialog::UpdatesAvailable() const
+    {
+        return !_pendingUpdateVersion.empty();
+    }
+
+    winrt::hstring AboutDialog::PendingUpdateVersion() const
+    {
+        return _pendingUpdateVersion;
+    }
+
+    void AboutDialog::_SetPendingUpdateVersion(const winrt::hstring& version)
+    {
+        _pendingUpdateVersion = version;
+        _PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"PendingUpdateVersion" });
+        _PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"UpdatesAvailable" });
+    }
+
+    winrt::fire_and_forget AboutDialog::QueueUpdateCheck()
+    {
+        auto strongThis = get_strong();
+        auto now{ std::chrono::system_clock::now() };
+        if (now - _lastUpdateCheck < std::chrono::days{ 1 })
+        {
+            co_return;
+        }
+        _lastUpdateCheck = now;
+
+        if (!IsPackaged())
+        {
+            co_return;
+        }
+
+        co_await wil::resume_foreground(strongThis->Dispatcher());
+        _SetPendingUpdateVersion({});
+        CheckingForUpdates(true);
+
+        try
+        {
+#ifdef WT_BRANDING_DEV
+            // **DEV BRANDING**: Always sleep for three seconds and then report that
+            // there is an update available. This lets us test the system.
+            co_await winrt::resume_after(std::chrono::seconds{ 3 });
+            co_await wil::resume_foreground(strongThis->Dispatcher());
+            _SetPendingUpdateVersion(L"X.Y.Z");
+#else // release build, likely has a store context
+            if (auto storeContext{ winrt::Windows::Services::Store::StoreContext::GetDefault() })
+            {
+                auto updates = co_await storeContext.GetAppAndOptionalStorePackageUpdatesAsync();
+                co_await wil::resume_foreground(strongThis->Dispatcher());
+                if (updates.Size() > 0)
+                {
+                    auto version = updates.GetAt(0).Package().Id().Version();
+                    _SetPendingUpdateVersion(fmt::format(L"{0}.{1}.{2}", version.Major, version.Minor, version.Revision));
```

Comment:
FYI this doesn't compile, because `std::wstring` and `winrt::hstring` aren't implicitly convertible.